games-arcade/rocksndiamonds/patches/rocksndiamonds-4.0.0.0_rc3.patchset

```diff
@@ -0,0 +1,34 @@
+From 315a9aa795538ad72b4be7b6540933e087faf719 Mon Sep 17 00:00:00 2001
+From: begasus <begasus@gmail.com>
+Date: Fri, 29 Jul 2016 13:59:37 +0200
+Subject: add -lnetwork and remove -lm from the source Makefile
+
+
+diff --git a/src/Makefile b/src/Makefile
+index dcc625c..bd5a14b 100644
+--- a/src/Makefile
++++ b/src/Makefile
+@@ -37,6 +37,11 @@ PROGNAME = ../$(PROGBASE)
+ # configuring platform
+ # -----------------------------------------------------------------------------
+ 
++ifeq ($(PLATFORM),haiku)		# native compiling to Haiku
++EXTRA_LDFLAGS = -lnetwork
++PLATFORM = unix
++endif
++
+ ifeq ($(PLATFORM),macosx)		# compiling on Mac OS X
+ EXTRA_LDFLAGS = -lstdc++
+ override PLATFORM = unix
+@@ -123,7 +128,7 @@ OPTIONS = -O3 -Wall
+ endif
+ 
+ CFLAGS = $(OPTIONS) $(SYS_CFLAGS)  $(EXTRA_CFLAGS) $(CONFIG)
+-LDFLAGS =           $(SYS_LDFLAGS) $(EXTRA_LDFLAGS) -lm
++LDFLAGS =           $(SYS_LDFLAGS) $(EXTRA_LDFLAGS)
```

Comment:
To be upstreamed, you should better define by default MATHLIBS to -lm, and set it to empty for Haiku. Then use MATHLIBS  in LDFLAGS.